src/pages/index.js

```diff
@@ -10,48 +12,131 @@ import Section from "../components/Section.js";
 
 import Card from "../components/Card.js";
 
+import Api from "../components/Api.js";
+
 import "./index.css";
 
-import { initialCards, config, variables } from "../utils/constants.js";
+import { config, variables } from "../utils/constants.js";
 
-// SECTION
+// API
 
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: (cardData) => {
-      createCard(cardData);
-    },
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "ebfbe580-59e8-4623-9d1e-5edf14608279",
+    "Content-Type": "application/json",
   },
-  "gallery__cards"
-);
+});
+
+// SECTION
 
-section.renderItems();
+let section;
+
+api
+  .getInitialCards()
+  .then((res) => {
+    section = new Section(
+      {
+        items: res,
+        renderer: (cardData) => {
+          createCard(cardData);
+        },
+      },
+      "gallery__cards"
+    );
+    section.renderItems();
+    console.log(res);
+  })
+  .catch(console.error);
+
+api
+  .loadInfo()
+  .then((res) => {
+    console.log(res);
+    userinfo.setUserInfo(res.name, res.about);
+    userinfo.setAvatar(res.avatar);
+  })
+  .catch(console.error);
 
 // POPUP WITH IMAGE
 
 const popupWithImage = new PopupWithImage("#picture-modal");
 
 popupWithImage.setEventListeners();
 
+// SAVE BUTTON TEXT CHANGE FUNCTION FOR FORMS
+
+function setButtonText(buttin, text) {
+  buttin.textContent = text;
+}
+
 // CARD
 
 function handleImageClick(cardData) {
   popupWithImage.open(cardData);
 }
 
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
   section.addItem(card.generateCard());
 }
 
 function handleAddCardSubmit(inputValues) {
-  createCard(inputValues);
-  variables.profileAddForm.reset();
-  addFormValidator.toggleButtonState();
+  setButtonText(variables.addModalButton, "Saving...");
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```